SDK_dotnet/GluwaClient.cs

```diff
@@ -0,0 +1,402 @@
+                }
+            }
+        }
+
+        public async Task<bool> PostTransactionAsync(
+            ECurrency? currency,
+            string amount,
+            string target,
+            string merchantOrderID = "",
+            string note = "",
+            int expiry = 1800
+            )
+        {
+            if (string.IsNullOrEmpty(amount))
+            {
+                throw new ArgumentNullException(nameof(amount));
+            }
+            else if (string.IsNullOrEmpty(target))
+            {
+                throw new ArgumentNullException(nameof(target));
+            }
+            else if (string.IsNullOrEmpty(getContractAddressOrNull(currency, mIsDevEnv)))
+            {
+                throw new ArgumentNullException(nameof(currency));
+            }
+
+            var requestUri = $"{mBaseUrl}/v1/Transactions";
+
+            var getFee = getFeeAsync(currency).Result.Data.MinimumFee;
+
+            BigInteger convertAmount = Converter.ConvertToGluwacoinBigInteger(amount);
+            BigInteger convertFee = Converter.ConvertToGluwacoinBigInteger(getFee);
+            int nonce = ((int)(DateTime.UtcNow - new DateTime(1970, 1, 1)).TotalSeconds);
+
+            ABIEncode abiEncode = new ABIEncode();
+            byte[] messageHash = abiEncode.GetSha3ABIEncodedPacked(
+                new ABIValue("address", getContractAddressOrNull(currency, mIsDevEnv)),
+                new ABIValue("address", mAddress),
+                new ABIValue("address", target),
+                new ABIValue("uint256", convertAmount),
+                new ABIValue("uint256", convertFee),
+                new ABIValue("uint256", nonce)
+                );
+
+            var signer = new EthereumMessageSigner();
+            string addressRecovered = signer.Sign(messageHash, mPrivateKey);
+
+            TransactionRequest bodyParams = new TransactionRequest
+            {
+                Signature = addressRecovered,
+                Currency = currency,
+                Target = target,
+                Amount = amount,
+                Fee = getFee,
+                Source = mAddress,
+                Nonce = nonce.ToString(),
+                MerchantOrderID = merchantOrderID,
+                Note = note
+            };
+
+            string json = bodyParams.ToJson();
+            StringContent content = new StringContent(json, Encoding.UTF8, "application/json");
+
+            using (HttpClient httpClient = new HttpClient())
+            using (var response = await httpClient.PostAsync(requestUri, content))
+            {
+                if (response.IsSuccessStatusCode)
+                {
+                    return true;
+                }
+                else
+                {
+                    throw new HttpListenerException((int)response.StatusCode, response.ReasonPhrase);
+                }
+            }
+        }
+
+        public async Task<Result<byte[]>> GetPaymentQRCodeAsync(
```

Comment:
This method should be in a separate class, say QrCodeClient.

There are two kinds of apis:
1. APIs that need api key and secret
2. APIs that don't need api key and secret. Basically, this is public apis

This method is 1., which means it doesn't belong with the rest of the APIs. If the user of this client only wants to use 2) and they don't have api key and secret, they can't use this SDK at all. So we need to separate it

Reply:
Separate the two with GluwaClient and QRCodeClient, and put them in the Clients folder. Is that right?